controllers/storagecluster/cephcluster.go

```diff
@@ -181,14 +181,17 @@ func (obj *ocsCephCluster) ensureCreated(r *StorageClusterReconciler, sc *ocsv1.
 				// reset the KMS connection's error field,
 				// it will be anyway set if there is an error
 				sc.Status.KMSServerConnection.KMSServerConnectionError = ""
-				if kmsConfigMap.Data["KMS_PROVIDER"] == "vault" {
-					sc.Status.KMSServerConnection.KMSServerAddress = kmsConfigMap.Data["VAULT_ADDR"]
-				} else if kmsConfigMap.Data["KMS_PROVIDER"] == AzureKSMProvider {
-					sc.Status.KMSServerConnection.KMSServerAddress = kmsConfigMap.Data["AZURE_VAULT_URL"]
-				} else if kmsConfigMap.Data["KMS_PROVIDER"] == ThalesKMSProvider {
-					sc.Status.KMSServerConnection.KMSServerAddress = kmsConfigMap.Data["KMIP_ENDPOINT"]
-				}
-				if err = reachKMSProvider(kmsConfigMap); err != nil {
+				// get the current KMS provider name
+				currKMSProvider := kmsConfigMap.Data[KMSProviderKey]
+				// according to the KMS provider, get the corresponding address/endpoint key
+				// and use it to set 'KMSServerAddress' status
+				kmsAddrssKey := kmsProviderAddressKeyMap[currKMSProvider]
+				sc.Status.KMSServerConnection.KMSServerAddress = kmsConfigMap.Data[kmsAddrssKey]
+				// if the KMS connection address is empty, log it as an error and continue
```

Comment:
@aruniiird you comment says to continue, Dont you want to continue to check the `reachKMSProvider(kmsConfigMap)`

Reply:
@iamniting ,
if the `KMSServerAddress` is empty, that means the endpoint address, which we are trying to reach, is empty, then we don't have to check/call `reachKMSProvider(kmsConfigMap)` function (as there is no address to check/reach).

> // if the KMS connection address is empty, log it as an error and continue

By logging error and continue, what I meant was to just print the error message (previously we didn't have any indication on why KMS is not working) and we continue to create the new CephCluster.

Modified the commit message, please check.